docs: add better examples and documentation (#391)
asyncapi/ts-nats-template: examples/README.md

# Examples

> This directory contains a series of self-contained examples that you can use as starting points, or as snippets to pull into your existing applications. All examples expect a NATS broker to  connect to.

Reviewer: ```suggestion
This directory contains a series of self-contained examples that you can use as starting points or snippets, to pull into your existing applications. All examples expect a NATS broker to connect to.
```

> The examples is as follows:

Reviewer: ```suggestion
The examples are as follows:
```

> - [simple-reply](./simple-reply) - A basic example of how to setup a reply in AsyncAPI and how to use the generated code.
> - [simple-request](./simple-request) - A basic example of how to send a request in AsyncAPI and how to use the generated code.

Reviewer: ```suggestion
- [simple-reply](./simple-reply) - A basic example of how to set up a reply in AsyncAPI and use the generated code.
- [simple-request](./simple-request) - A basic example of how to send a request in AsyncAPI and use the generated code.
```
